lighthouse-core/audits/autocomplete.js

```diff
@@ -43,22 +194,80 @@ class AutocompleteAudit extends Audit {
     };
   }
 
+  /**
+   * @param {LH.Artifacts.FormInput} input
+   * @return {{hasValidTokens: Boolean, inValidOrder?: Boolean}}
+   */
+  static checkAttributeValidity(input) {
+    if (!input.autocomplete.attribute) return {hasValidTokens: false};
+    const tokenArray = input.autocomplete.attribute.split(' ');
+    for (const token of tokenArray) {
+      // A `section-` prefix indicates a unique autofill scope.
+      // https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#:~:text=section-
+      if (token.slice(0, 8) === 'section-') continue;
+      if (validAutocompleteTokens.includes(token)) continue;
+      return {hasValidTokens: false};
+    }
+    // If all autocomplete tokens are valid but there is still no property attribute, then that means the tokens are out of order.
+    // https://cloudfour.com/thinks/autofill-what-web-devs-should-know-but-dont/#all-the-tokens
+    if (!input.autocomplete.property) return {hasValidTokens: true, inValidOrder: true};
+    return {hasValidTokens: true};
+  }
+
   /**
    * @param {LH.Artifacts} artifacts
    * @return {LH.Audit.Product}
    */
   static audit(artifacts) {
     const forms = artifacts.FormElements;
     const failingFormsData = [];
+    const warnings = [];
+    let notApplicable = 0;
+    let inputsCount = 0;
     for (const form of forms) {
       for (const input of form.inputs) {
-        if (!input.autocomplete) {
+        inputsCount += 1;
+        const autocomplete = this.checkAttributeValidity(input);
+        if (!autocomplete.hasValidTokens || autocomplete.inValidOrder) {
+          if (!input.autocomplete.prediction) {
+            notApplicable += 1;
+            continue;
+          }
+          if (noPrediction.includes(input.autocomplete.prediction) &&
+          !input.autocomplete.attribute) {
+            notApplicable += 1;
+            continue;
+          }
+          const snippetArray = input.snippet.split(' title=');
```

Comment:
What is this doing? What if there is no `title`?

Reply:
Ill add a comment. this is taking away the autofill injection that the flag adds to the DOM. basically making it easier for developers to read the snippet.

Reply:
makes sense we can exclude those from the snippet over in the gatherer instead


```diff
diff --git a/lighthouse-core/gather/gatherers/form-elements.js b/lighthouse-core/gather/gatherers/form-elements.js
index e54d482d6..bbd4b5ce1 100644
--- a/lighthouse-core/gather/gatherers/form-elements.js
+++ b/lighthouse-core/gather/gatherers/form-elements.js
@@ -24,6 +24,7 @@ function collectFormElements() {
     inputs: [],
     labels: [],
   };
+  const snippetIgnoreAttrs = ['autofill-information', 'autofill-prediction', 'title'];
   for (const child of formChildren) {
     const isButton = child instanceof HTMLInputElement &&
       (child.type === 'submit' || child.type === 'button');
@@ -40,7 +41,7 @@ function collectFormElements() {
           // @ts-expect-error - put into scope via stringification
           nodeLabel: getNodeLabel(parentFormElement), // eslint-disable-line no-undef,
           // @ts-expect-error - put into scope via stringification
-          snippet: getOuterHTMLSnippet(parentFormElement), // eslint-disable-line no-undef
+          snippet: getOuterHTMLSnippet(parentFormElement, snippetIgnoreAttrs), // eslint-disable-line no-undef
         },
         inputs: [],
         labels: [],
@@ -63,7 +64,7 @@ function collectFormElements() {
         // @ts-expect-error - put into scope via stringification
         nodeLabel: getNodeLabel(child), // eslint-disable-line no-undef,
         // @ts-expect-error - put into scope via stringification
-        snippet: getOuterHTMLSnippet(child), // eslint-disable-line no-undef
+        snippet: getOuterHTMLSnippet(child, snippetIgnoreAttrs), // eslint-disable-line no-undef
       });
     }
     if (child instanceof HTMLLabelElement) {

```